src/modules/ce/components/referral/ReferralDetailContent.tsx

```diff
@@ -1,85 +1,243 @@
 import { Stack } from '@mui/system';
 import { isNil } from 'lodash-es';
-import CommonCard from '@/components/elements/CommonCard';
-import CommonDetailGrid from '@/components/elements/CommonDetailGrid';
+import { useMemo, useState } from 'react';
+import CommonCollapsibleCard from '@/components/elements/CommonCollapsibleCard';
+import CommonDetailGrid, {
+  CommonDetailGridItemRow,
+} from '@/components/elements/CommonDetailGrid';
+import ExternalIdDisplay from '@/components/elements/ExternalIdDisplay';
 import RouterLink from '@/components/elements/RouterLink';
 import ReferralStatusChip from '@/modules/ce/components/referral/ReferralStatusChip';
 import {
   clientNameFromRecordWithOptionalClient,
+  entryExitRange,
   parseAndFormatDate,
 } from '@/modules/hmis/hmisUtil';
-import { ClientDashboardRoutes } from '@/routes/routes';
-import { CeReferralFieldsFragment } from '@/types/gqlTypes';
+import {
+  ClientDashboardRoutes,
+  EnrollmentDashboardRoutes,
+} from '@/routes/routes';
+import { HmisEnums } from '@/types/gqlEnums';
+import {
+  CeReferralFieldsFragment,
+  ExternalIdentifierType,
+} from '@/types/gqlTypes';
 import { generateSafePath } from '@/utils/pathEncoding';
 
 interface Props {
   referral: CeReferralFieldsFragment;
 }
 
 const ReferralDetailContent: React.FC<Props> = ({ referral }) => {
+  const [cardStates, setCardStates] = useState({
+    referralDetails: true,
+    clientDetails: true,
+    sourceEnrollmentDetails: false,
+    eligibilityRequirementsDetails: false,
+  });
+
+  const toggleCard = (cardKey: keyof typeof cardStates) => {
+    setCardStates((prev) => ({
+      ...prev,
+      [cardKey]: !prev[cardKey],
+    }));
+  };
+
+  const referralDetails = useMemo(
+    () => [
+      {
+        id: 'status',
+        label: 'Referral Status',
+        value: <ReferralStatusChip status={referral.status} size='small' />,
+      },
+      { id: 'id', label: 'Referral ID', value: referral.id },
+      {
+        id: 'referralDate',
+        label: 'Date Started',
+        value: parseAndFormatDate(referral.createdAt),
+      },
+      {
+        id: 'unit',
+        label: 'Unit',
+        value: referral.opportunity.name,
+      },
+      {
+        id: 'project',
+        label: 'Project',
+        value: referral.targetProjectName,
+      },
+    ],
+    [referral]
+  );
+
+  const clientDetails = useMemo(() => {
+    const rows: CommonDetailGridItemRow[] = [
+      { id: 'clientId', label: 'Client ID', value: referral.clientId },
+    ];
+
+    if (referral.client) {
+      rows.push({
+        id: 'clientName',
+        label: 'Client Name',
+        value: (
+          <RouterLink
+            to={generateSafePath(ClientDashboardRoutes.PROFILE, {
+              clientId: referral.clientId,
+            })}
+            openInNew
+          >
+            {clientNameFromRecordWithOptionalClient(referral)}
```

Comment:
I think we can just use `clientBriefName(referral.client)` here since we know `referral.client` is non-null

Reply:
Also, based on this code, it looks like client name and age won't be shown unless the `referral.client` is present (user has permission to view), which may be unintentional, based on [backend field descriptions](https://github.com/greenriver/hmis-warehouse/pull/5540/files#diff-4e19db10b1ce14bc2a4ee04b574b3e7d69467f2482560c939698bff9116929b4R18-R19)?

Reply:
https://github.com/greenriver/hmis-frontend/pull/1196/commits/d5d319e7b3bd5aca016a56ab710bf1a0028cfb86